### ISSUE TITLE
Configure firewall for Dovecot: 'dict object' has no attribute 'login-service'

### ISSUE THREAD

**Issue by ganto** [debops/ansible-dovecot#13]
When defining some custom values in a ``dovecot_<protocol>_config_map`` e.g.:

    dovecot_imap_config_map:
      protocol:
        mail_plugins: '$mail_plugins imap_quota'

but don't define a custom ``login-service`` listener, the ferm task will fail with the following error:

    TASK: [debops.dovecot | Configure firewall for Dovecot] *********************** 
    fatal: [mail01.example.com] => {'msg': "AnsibleUndefinedVariable: One or more undefined variables: 'dict object' has no attribute 'login-service'", 'failed': True}
    fatal: [mail01.example.com] => {'msg': "AnsibleUndefinedVariable: One or more undefined variables: 'dict object' has no attribute 'login-service'", 'failed': True}

This is caused by the rather hack-ish dict lookup done in [etc/ferm/filter-input.d/dovecot.conf.j2](https://github.com/debops/ansible-dovecot/blob/3a6773a27079b7aef9b76b86e8c2626cf8b13a93/templates/etc/ferm/filter-input.d/dovecot.conf.j2#L30).

**Workaround**

Just define a minimal ``login-service`` listener, e.g.:
    
    dovecot_imap_config_map:
      protocol:
        mail_plugins: '$mail_plugins imap_quota'
      login-service:
        inet_listener:
          imap:
            allow: []

**TODO**

Ah, I knew that this would bite me one day... Does someone know a nice(r) way to lookup an optional value in a nested dict?

**Comment by drybjed** [comment]
The only clean way I know of is to check if a dict key you want is defined in that dict, no other way around it, something like:
```
{% if dovecot_map_config_map['login-service']|d() %}
# do the things with that dict key
{% endif %}
```
You could for example rebuild the data inside the template based on what is in Ansible inventory merged with what you know about the defaults, and derive the configuration from that local data in the template.

**Comment by ganto** [comment]
Shut be fixed now after merging #14
